setup.py

```diff
@@ -33,6 +33,7 @@
       url='http://code.google.com/p/pyfpdf',
       license='LGPLv3+',
       download_url="https://github.com/reingart/pyfpdf/tarball/%s" % fpdf.__version__,
+      install_requires=['numpy>=1.15.4', 'Pillow>=5.3.0'],
```

Comment:
Disagree with numpy depency

Reply:
Please have a look here:

https://github.com/reingart/pyfpdf/blob/08e28af35b686a67f777113ec571ac6d278d51f1/fpdf/fpdf.py#L1945

Here, bytes representing RGBA values are processed using regular expressions. Though this is a very creative solution, it is extremely slow.

Reply:
Are you referring to this code?:

https://github.com/alexanderankin/pyfpdf/blob/master/fpdf/image_parsing.py#L199

In any case, the way the pixels are being processed (by means of regular expressions seperating RGBA into RGB and A) is really really slow. So, yes, numpy is an additional dependency, but at least image handling is much more performant now.

Reply:
oh im saying ill accept such a pr on my fork thats all. i agree with this move.

Reply:
actually can you help me test it? I copied this bit into my branch and the way i have my tests set up is that it generates a pdf, computes hash, compares the hash and then `os.unlink`s the generated pdf. However, with this proposed `get_img_info` function it is inserting an extra object and stream for the thumbnail preview, which of course makes the test fail but i should probably just re-do all my tests? thoughts?

Reply:
i updated mine to 2.0.1 with all these changes but it broke a lot of my tests :/

Reply:
I pushed another fix to this branch to properly deal with unsupported extensions. Not sure about the inner workings of the test suite, but it seems to me that due to the Pillow changes the hashes of the resources changed. 

```
>> Tests: 28
Test 1 / 28 : test_cache.py
NORES   NORES   NORES   NORES   NORES   NORES   NORES   NORES   NORES   NORES   NORES   NORES   NORES   
Test 2 / 28 : test_corebox.py
NORES   NORES   NORES   NORES   NORES   NORES   NORES   NORES   NORES   NORES   NORES   NORES   NORES   
Test 3 / 28 : test_e1252.py
OK      OK      OK      OK      OK      OK      OK      OK      OK      OK      OK      OK      OK      
Test 4 / 28 : test_imgmask.py
HASHER  SKIP    SKIP    SKIP    SKIP    SKIP    SKIP    SKIP    SKIP    SKIP    SKIP    SKIP    SKIP    
Test 5 / 28 : test_invoice.py
HASHER  FAIL    FAIL    FAIL    FAIL    FAIL    FAIL    FAIL    FAIL    FAIL    FAIL    FAIL    FAIL    
Test 6 / 28 : test_issue14.py
HASHER  FAIL    FAIL    FAIL    FAIL    FAIL    FAIL    FAIL    FAIL    FAIL    FAIL    FAIL    FAIL    
Test 7 / 28 : test_issue33.py
HASHER  SKIP    SKIP    SKIP    SKIP    SKIP    SKIP    SKIP    SKIP    SKIP    SKIP    SKIP    SKIP    
Test 8 / 28 : test_issue35.py
OK      OK      OK      OK      OK      OK      OK      OK      OK      OK      OK      OK      OK      
Test 9 / 28 : test_issue41.py
OK      OK      OK      OK      OK      OK      OK      OK      OK      OK      OK      OK      OK      
Test 10 / 28 : test_issue60.py
SKIP    SKIP    SKIP    SKIP    SKIP    SKIP    SKIP    SKIP    SKIP    SKIP    SKIP    SKIP    SKIP    
Test 11 / 28 : test_issue62.py
NORES   NORES   NORES   NORES   NORES   NORES   NORES   NORES   NORES   NORES   NORES   NORES   NORES   
Test 12 / 28 : test_issue63.py
NORES   NORES   NORES   NORES   NORES   NORES   NORES   NORES   NORES   NORES   NORES   NORES   NORES   
Test 13 / 28 : test_issue70.py
OK      OK      OK      OK      OK      OK      OK      OK      OK      OK      OK      OK      OK      
Test 14 / 28 : test_issue71.py
OK      OK      OK      OK      OK      OK      OK      OK      OK      OK      OK      OK      OK      
Test 15 / 28 : test_issue78.py
HASHER  SKIP    SKIP    SKIP    SKIP    SKIP    SKIP    SKIP    SKIP    SKIP    SKIP    SKIP    SKIP    
Test 16 / 28 : test_issue82.py
NORES   NORES   NORES   NORES   NORES   NORES   NORES   NORES   NORES   NORES   NORES   NORES   NORES   
Test 17 / 28 : test_jpeg.py
OK      OK      OK      OK      OK      OK      OK      OK      OK      OK      OK      OK      OK      
Test 18 / 28 : test_nbpages.py
NORES   NORES   NORES   NORES   NORES   NORES   NORES   NORES   NORES   NORES   NORES   NORES   NORES   
Test 19 / 28 : test_output.py
OK      OK      OK      OK      OK      OK      OK      OK      OK      OK      OK      OK      OK      
Test 20 / 28 : test_page_orient.py
OK      OK      OK      OK      OK      OK      OK      OK      OK      OK      OK      OK      OK      
Test 21 / 28 : test_page_size.py
OK      OK      OK      OK      OK      OK      OK      OK      OK      OK      OK      OK      OK      
Test 22 / 28 : test_py3k.py
OK      OK      OK      OK      OK      OK      SKIP    SKIP    OK      SKIP    SKIP    SKIP    SKIP    
Test 23 / 28 : test_simple.py
HASHER  SKIP    SKIP    SKIP    SKIP    SKIP    SKIP    SKIP    SKIP    SKIP    SKIP    SKIP    SKIP    
Test 24 / 28 : test_stretching.py
NORES   NORES   NORES   NORES   NORES   NORES   NORES   NORES   NORES   NORES   NORES   NORES   NORES   
Test 25 / 28 : test_template.py
OK      OK      OK      OK      OK      OK      OK      OK      OK      OK      OK      OK      OK      
Test 26 / 28 : test_ttfonts.py
NORES   NORES   NORES   NORES   NORES   NORES   NORES   NORES   NORES   NORES   NORES   NORES   NORES   
Test 27 / 28 : test_unicode.py
NORES   NORES   NORES   NORES   NORES   NORES   NORES   NORES   NORES   NORES   NORES   NORES   NORES   
Test 28 / 28 : test_winfonts.py
SKIP    SKIP    SKIP    SKIP    SKIP    SKIP    SKIP    SKIP    SKIP    SKIP    SKIP    SKIP    SKIP    
```